tdi_python/tdicli.py

```diff
@@ -50,11 +51,15 @@
 ipython_app = None
 
 class CIntfTdi:
+    target_type_cls = TargetType
 
-    def __init__(self, dev_id, table_cls, info_cls):
+    def __init__(self, dev_id, table_cls, info_cls, driver_path, utils_path):
+    #def __init__(self, dev_id, table_cls, info_cls):
         self._dev_id = dev_id
-        self._utils = CDLL(install_directory+'/lib/libtarget_utils.so', mode=RTLD_GLOBAL)
-        self._driver = CDLL(install_directory+'/lib/libdriver.so')
+        self._utils = CDLL(utils_path, mode=RTLD_GLOBAL)
+        self._driver = CDLL(driver_path)
+        # self._utils = CDLL(install_directory+'/lib/libtarget_utils.so', mode=RTLD_GLOBAL)
+        # self._driver = CDLL(install_directory+'/lib/libdriver.so')
```

Comment:
Please move the functions: _set_pipe, _set_direction, _set_parser to target specific (line 223-230).
The constructor: self._dev_tgt = self.TdiDevTgt(self._dev_id, 0, 0xff, 0xff) to target specific too.

Reply:
Will remove them in next phase of refactoring work